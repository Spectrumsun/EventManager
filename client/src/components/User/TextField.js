import React from 'react';

const TextField = ({
  type, value, onChange, name, label, placeholder
}) => (
  <div className="form-group">
    <label htmlFor="2">{label}</label>
    <input
      type={type}
      value={value}
      onChange={onChange}
      name={name}
      className="form-control form-control-lg"
      placeholder={placeholder}
      //required
    />
  </div>
);


export default TextField;
