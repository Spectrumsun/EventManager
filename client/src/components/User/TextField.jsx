import React from 'react';
import PropTypes from 'prop-types'; 

const TextField = ({
  type, value, onChange, name, label, placeholder
}) => (
  <div className="form-group">
    <label htmlFor="2">{label}</label>
    <input
      type={type}
      value={value}
      onChange={onChange}
      name={name}
      className="form-control form-control-lg"
      placeholder={placeholder}
    
    />
  </div>
);


TextField.propTypes = {
  type: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  name: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  placeholder: PropTypes.string.isRequired
};

export default TextField;
