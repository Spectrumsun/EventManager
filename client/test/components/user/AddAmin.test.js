import React from 'react';
import { mount, shallow } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import render from 'react-test-renderer';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { createMemoryHistory } from 'history';
import sinon from 'sinon';
import ConnectedAddAmin,
{ AddAmin, mapDispatchToProps } from '../../../src/components/User/addAmin';
import TextField from '../../../src/components/User/TextField'

const middleware = [thunk];
const mockStore = configureStore(middleware);
const initialState = {
  isAuthenticated: false,
  user: {},
};
const store = mockStore(initialState);

const props = {
  initaddAdmin: sinon.spy(() => new Promise((cb) => {
    cb();
  })),
  history: createMemoryHistory()
};

const mountedWrapper = mount(
  <Provider store={store}>
  <BrowserRouter>
    <ConnectedAddAmin {...props} />
  </BrowserRouter>
  </Provider>
);

const shallowWrapper = shallow(<AddAmin {...props} />);

const state = {
  email: 'bot@yahoo.com',
  role: 'user',
};

const event = {
  preventDefault: jest.fn(),
  target: {
    email: 'bot@yahoo.com',
    role: 'user',
  }
};

// let wrapper;

describe('<Admin /> Component', () => {
  it('should render the <Login />', () => {
    shallow(<AddAmin />);
  });

  it('should render the <Admin /> without crashing', () => {
    expect(mountedWrapper).toBeDefined();
    expect(mountedWrapper.find('AddAmin').length).toBe(1);
  });

  it('should have <TextField /> when the page loads', () => {
    const wrapper = shallow(<AddAmin />);
    expect(wrapper.find(TextField)).toHaveLength(2);
  });

  it('should match component snapshot', () => {
    const tree = render.create(
      <BrowserRouter >
        <AddAmin {...props} />
      </BrowserRouter>);
    expect(tree).toMatchSnapshot();
  });

  it('should have div on layout', () => {
    const wrapper = shallow(<AddAmin />);
    expect(wrapper.find('button').length).toEqual(1);
  });

  it('calls onChange event when input is passed to state', () => {
    sinon.spy(shallowWrapper.instance(), 'onChange');
    shallowWrapper.instance().onChange(event);
    expect(shallowWrapper.instance().onChange.calledOnce).toEqual(true);
  });

  it('calls onSubmit event when submit button is clicked', () => {
    sinon.spy(shallowWrapper.instance(), 'onSubmit');
    shallowWrapper.setState(state);
    shallowWrapper.instance().onSubmit(event);
    expect(shallowWrapper.instance().onSubmit.calledOnce).toEqual(true);
  });

  it('ensures that mapDispatchToProps dispatches the specified actions', () => {
    const dispatch = jest.fn();
    expect(mapDispatchToProps(dispatch).initaddAdmin).toBeTruthy();
  });

});

