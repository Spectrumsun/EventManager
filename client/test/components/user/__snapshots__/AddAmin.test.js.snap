// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Admin /> Component should match component snapshot 1`] = `
<div>
  <div
    className="container"
    style={
      Object {
        "paddingTop": "150px",
      }
    }
  >
    <h3
      className="center"
    >
      Event Manager
    </h3>
    <div
      className="card loginCard"
      style={
        Object {
          "width": "30rem",
        }
      }
    >
      <div
        className="card-header"
      >
        <h3>
          Add New Admin
        </h3>
      </div>
      <div
        className="card-body"
      >
        <div
          className="cont card-body"
        >
          <form
            className="centerform"
            onSubmit={[Function]}
          >
            <div
              className="form-group"
            >
              <label
                htmlFor="2"
              >
                Email
              </label>
              <input
                className="form-control form-control-lg"
                name="email"
                onChange={[Function]}
                placeholder="Vaild Email"
                required={true}
                type="email"
                value=""
              />
            </div>
            <div
              className="form-group"
            >
              <label
                htmlFor="2"
              >
                Role
              </label>
              <input
                className="form-control form-control-lg"
                name="role"
                onChange={[Function]}
                placeholder="role"
                required={true}
                type="text"
                value=""
              />
            </div>
            <div
              className="text-center"
            >
              <button
                className="btn btn-outline-dark"
                type="submit"
              >
                Submit
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
